datafusion/optimizer/src/decorrelate_where_in.rs

```diff
@@ -263,16 +344,15 @@ mod tests {
             .build()?;
 
         let expected = "Projection: test.b [b:UInt32]\
-        \n  LeftSemi Join: test.b = __correlated_sq_2.c [a:UInt32, b:UInt32, c:UInt32]\
-        \n    LeftSemi Join: test.c = __correlated_sq_1.c [a:UInt32, b:UInt32, c:UInt32]\
+        \n  LeftSemi Join:  Filter: test.b = __correlated_sq_2.c [a:UInt32, b:UInt32, c:UInt32]\
```

Comment:
I believe these changes mean that the plans expressions are no no longer in the `ON` (equijoin) clause but instead are in the 


However I think this is fine as other optimizer passes will move such predicates into the on clause. I tested this locally with this branch 👍 

```sql
❯ explain select *  from '/tmp/data.csv' t where t.name IN (select name from '/tmp/data.csv' where name = 'Sales');
+---------------+-----------------------------------------------------------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                                                                    |
+---------------+-----------------------------------------------------------------------------------------------------------------------------------------+
| logical_plan  | Projection: t.name, t.created_at, t.last_report                                                                                         |
|               |   LeftSemi Join: t.name = __correlated_sq_2.name                                                                                        |
|               |     SubqueryAlias: t                                                                                                                    |
|               |       TableScan: /tmp/data.csv projection=[name, created_at, last_report]                                                               |
|               |     SubqueryAlias: __correlated_sq_2                                                                                                    |
|               |       Projection: /tmp/data.csv.name AS name                                                                                            |
|               |         Filter: /tmp/data.csv.name = Utf8("Sales")                                                                                      |
|               |           TableScan: /tmp/data.csv projection=[name], partial_filters=[/tmp/data.csv.name = Utf8("Sales")]          
```